src/AutoUI/schemaOps.ts

```diff
@@ -36,6 +37,13 @@ export interface AutoUIModel<T> {
 	priorities?: Priorities<T>;
 }
 
+export type AutoUITagsSdk<T> =
+	| (ResourceTagModelService & {
+			getAll: (itemsOrFilters: any) => T[] | Promise<T[]>;
+			canAccess: () => Promise<boolean>;
+	  })
+	| ResourceTagModelService;
```

Comment:
Afaict, since we are not using these as discriminating properties for the union, this could be just:
```suggestion
export type AutoUITagsSdk<T> = ResourceTagModelService & {
	getAll?: (itemsOrFilters: any) => T[] | Promise<T[]>;
	canAccess?: () => Promise<boolean>;
};
```
can't it?

Reply:
Actually it can only be one or the other 
- only submit ( in a non server side paginated case ) 
- all three are needed in a server side paginated world

Reply:
Then something like v ?
```suggestion
export type AutoUITagsSdk<T> = ResourceTagModelService & ({} | {
	getAll: (itemsOrFilters: any) => T[] | Promise<T[]>;
	canAccess: () => Promise<boolean>;
});
```

Reply:
this is exactly the same thing, but, IMHO semantically speaking, the inclusion of an empty object in the union can be confusing, implying that there's a significant optionality or variability in the type that doesn't actually exist. It doesn't clearly communicate the intent that getAll and canAccess are either both present or not present.

Having said that, since these express the same exact concept, if you prefer that way I'll apply it.